### DISCUSSION TITLE
Installing on Windows, get "error LNK2001"

### DISCUSSION THREAD

**Discussion by famfile** [bigcat88/pillow_heif#197] [Q&A] [answered]
I installed libheif and files are located in C:\msys64\mingw64[\bin\lib\etc.]. However, when I try to install pillow_heif, it crashes at this point:

```
Creating library build\temp.win32-cpython-312\Release\pillow_heif\_pillow_heif.cp312-win32.lib and object build\temp.win32-cpython-312\Release\pillow_heif\_pillow_heif.cp312-win32.exp
      _pillow_heif.obj : error LNK2001: unresolved external symbol __imp__heif_encoder_descriptor_get_name
      _pillow_heif.obj : error LNK2001: unresolved external symbol __imp__heif_image_handle_release
      _pillow_heif.obj : error LNK2001: unresolved external symbol __imp__heif_image_handle_get_luma_bits_per_pixel
      _pillow_heif.obj : error LNK2001: unresolved external symbol __imp__heif_get_encoder_descriptors
      _pillow_heif.obj : error LNK2001: unresolved external symbol __imp__heif_nclx_color_profile_free
```
Not sure what I'm doing wrong. Thoughts?

**Comment by bigcat88** [comment]
what version of libheif you install with MSYS, the last one I hope?

And do libheif is installed via default MSYS PKGBUILD?

**Reply by famfile** [reply]
Yes, I have installed it via PKGBUILD, version mingw-w64-x86_64-libheif-1.17.6-2. I tried again today to no avail. Still cannot get pillow_heif to install. I checked and ensured MSYS2_PREFIX is set properly ("C:\msys64\mingw64") as well.

```
$ pacman -U mingw-w64-x86_64-libheif-1.17.6-2-any.pkg.tar.zst
loading packages...
warning: mingw-w64-x86_64-libheif-1.17.6-2 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) mingw-w64-x86_64-libheif-1.17.6-2

Total Installed Size:  3.37 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] y
(1/1) checking keys in keyring                               [###############################] 100%
(1/1) checking package integrity                             [###############################] 100%
(1/1) loading package files                                  [###############################] 100%
(1/1) checking for file conflicts                            [###############################] 100%
(1/1) checking available disk space                          [###############################] 100%
:: Processing package changes...
(1/1) reinstalling mingw-w64-x86_64-libheif                  [###############################] 100%
```

**Comment by bigcat88** [comment]
Can you share full list of commands how to reproduce this?

**Reply by famfile** [reply]
Before the full list, I did have one issue (which I ignored). I followed the instructions [here](https://pillow-heif.readthedocs.io/en/latest/installation.html). It states to do the following:
```
cd .../pillow_heif/libheif/windows/mingw-w64-libheif
makepkg-mingw --syncdeps
pacman -U mingw-w64-x86_64-libheif-*-any.pkg.tar.zst
```
However, I had an issue with `makepkg-mingw --syncdeps` ... could this be the issue? See below.

```
$ makepkg-mingw --syncdeps
=> WARNING: MINGW_ARCH not set and not called from a MINGW environment, defaulting to mingw64. This will fail in the future!
==>  MINGW_ARCH: mingw64
  -> Building mingw64...
==> ERROR: PKGBUILD does not exist.
```

**Reply by bigcat88** [reply]
"ERROR: PKGBUILD does not exist."
maybe you were in the wrong directory?
`makepkg-mingw --syncdeps` should be done inside this directory:
https://github.com/bigcat88/pillow_heif/tree/master/libheif/windows/mingw-w64-libheif

But we are mixing two types of installation together, let's focus on one.
1. Is Python used as a Windows version or from the MSYS2-mingw package?
2. I assume that Python is 64 bit? Can I see the full path to python interpreter?

**Comment by famfile** [comment, marked as answer]
Well, apparently, I had both Python 32 bit and 64 bit installed; and the default was 32 bit. After adjusting that, python_heif installed without issue. Thank you!
